src/pkg/fault/fault.go

```diff
@@ -194,6 +205,35 @@ func (e *Bus) addSkip(s *Skipped) *Bus {
 	return e
 }
 
+// AddAlert appends a record of an Alert message to the fault bus.
+// Importantly, alerts are not errors, exceptions, or skipped items.
+// An alert should only be generated if no other fault functionality
+// is in use, but that we still want the end user to clearly and
+// plainly receive a notification about a runtime event.
+func (e *Bus) AddAlert(ctx context.Context, a *Alert) {
+	if a == nil {
+		return
+	}
+
+	e.mu.Lock()
```

Comment:
Should we move the locking to `addAlert`?

Reply:
Hmm, will follow up in a different PR.  This has some thorny side effects that I need to even out.